app/src/main/kotlin/compiler/lowlevel/allocation/Allocation.kt

```diff
@@ -10,6 +10,25 @@ import compiler.lowlevel.dataflow.Liveness
 
 object Allocation {
 
+    val REGISTER_ORDER = listOf(
+        Register.R15,
+        Register.R14,
+        Register.R13,
+        Register.R12,
+        Register.R11,
+        Register.R10,
+        Register.RBP,
+        Register.RSP,
+        Register.R9,
+        Register.R8,
+        Register.RDI,
+        Register.RSI,
+        Register.RDX,
+        Register.RCX,
+        Register.RBX,
+        Register.RAX,
+    )
```

Comment:
Regarding your concerns about RSP being used in allocation, wouldn't removing it from here solve the problem?

Reply:
Speaking of which, I think RBP should be also excluded from here (at least for now). While it is not that important as RSP (programs should still work correctly), the debugger (which might be useful for us) will have problems (for example with showing a stack trace).

Reply:
Also, I think the order of these registers is not optimal. The comment in Allocation.kt:38 says "Registers should be sorted by likeliness to be reserved for spilled". R15, R14, R13 are all callee-saved registers, so we should use them for spills last, because we read spilled values from memory every time, so conserving value of spill-handling registers across function calls is pointless, while it's useful for local variables.
I would suggest moving R11, R10 to the top of that list, as these registers are caller-saved and they are not used in parameter passing (which could potentially cause some problems).

Reply:
> Regarding your concerns about RSP being used in allocation, wouldn't removing it from here solve the problem?

OK, I see that it won't fix it, because RSP is being then colored by some other register.